### PR TITLE
feat: 🎸 Make `currentFundingRound` return `null` when empty

### PR DESCRIPTION
**Pull request by prashantasdeveloper**
BREAKING CHANGE: 🧨 `currentFundingRound` now returns null for empty funding round, thus
changing the return type `Promise<string | null>`

### CODE REVIEW

**Review by polymath-eric — APPROVED** [review]

File: src/utils/conversion.ts [hunk 1]
```diff
@@ -1601,8 +1601,8 @@ export function stringToFundingRoundName(roundName: string, context: Context): F
 /**
  * @hidden
  */
-export function fundingRoundNameToString(roundName: FundingRoundName): string {
-  return roundName.toString();
+export function fundingRoundNameToString(roundName: FundingRoundName): string | null {
```
Comment (monitz87) [inline src/utils/conversion.ts:1604]: I'd rather you perform the isEmpty check in `currentFoundingRound` and leave this as it was. It'll be easier to adapt when core changes the return type to `Option`

**Review by monitz87 — CHANGES_REQUESTED** [review]

File: src/api/entities/SecurityToken/index.ts [hunk 2]
```diff
@@ -320,12 +320,12 @@ export class SecurityToken extends Entity<UniqueIdentifiers, string> {
...
     if (callback) {
       return asset.fundingRound(rawTicker, round => {
-        callback(fundingRoundNameToString(round));
+        callback(fundingRoundNameToString(round) || null);
       });
     }
 
     const fundingRound = await asset.fundingRound(rawTicker);
-    return fundingRoundNameToString(fundingRound);
+    return fundingRoundNameToString(fundingRound) || null;
```
Comment (monitz87) [inline src/api/entities/SecurityToken/index.ts:329]: This behavior should probably be encapsulated into a function

```
const assembleResult = (roundName: FundingRoundName): string | null => fundingRoundNameToString(roundName) || null;

if (callback) {
  return asset.fundingRound(rawTicker, round => {
    callback(assembleResult(round));
  });
}

const fundingRound = await asset.fundingRound(rawTicker);
return assembleResult(fundingRound);
```

Easier to change when we move to Options, less prone to bugs

**Review by monitz87 — CHANGES_REQUESTED** [review]

File: src/api/entities/SecurityToken/index.ts (hunk 2 above)
Comment (prashantasdeveloper) [reply src/api/entities/SecurityToken/index.ts:329]: I infact did that and then reverted it 😅

**Review by monitz87 — APPROVED** [review]

**Pull request merged**